src/main/java/com/google/ehub/data/CommentDataManager.java

```diff
@@ -52,10 +61,17 @@ public void addItemComment(long itemId, String comment, long timestampMillis) {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery comments = datastore.prepare(itemCommentQuery);
     for (Entity entity : comments.asIterable()) {
+      String email = (String) entity.getProperty(EMAIL_PROPERTY_KEY);
+      UserProfile userProfile = profileData.getUserProfile(email);
+      if(userProfile == null){
```

Comment:
I'm glad you put a null check. It would prevent the item page servlet from breaking.